spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -225,7 +211,51 @@ data class SpdxLicenseExceptionExpression(
     override fun toString() = id
 }
 
-sealed class SpdxSimpleExpression : SpdxExpression()
+sealed class SpdxSingleLicenseExpression : SpdxExpression()
+
+data class SpdxLicenseWithExceptionExpression(
+    val license: SpdxSimpleExpression,
+    val exception: SpdxLicenseExceptionExpression
```

Comment:
The last time (while ago) I checked SPDX spec. the exception could only be a SPDX exception identifier, as implemented. This `imo` is a problem with the SPDX spec. as it makes it a bit more difficult  to properly deal with exceptions which do not have a valid identifier assigned in SPDX (yet). E.g.: 
`GPL-2.0-only with LicenseRef-scancode-some-exception' would not be possible and thus the only way to work around is to define a custom `LicenseRef` denoting a license-exception-combination like: `LicenseRef-$yourorg-gpl-2.0-only-with-scancode-some-exception`.

Would it make sense to deviate from the spec in this regard? (Also it may be interesting to how this use case is addressed in the next SPDX version)

What do you think?

Reply:
As discussed offline, let's keep it as is for now and discuss with @tsteenbe if there are plans for SPDX 3 to support multiple exceptions.